From b50a33617f8bfe0e2439c96b8cb6ccf70627fcaa Mon Sep 17 00:00:00 2001
From: Yu Feng <rainwoodman@gmail.com>
Date: Wed, 13 Jun 2018 17:26:52 -0700
Subject: [PATCH] define SPEED_UNKNOWN for compilation with anaconda compilers

When compiling torch with anaconda compilers I ran into this error about SPEED_UNKNOWN undefined:

https://github.com/pytorch/pytorch/issues/4499

I couldn't find the proper header to include -- even if found it is likely not portable.
Given that not too many people outside kernel uses this macro (gloo is one of the few results googling with 'SPEED_KNOWN'), it is
probably a good idea to define it if it is not defined on the system.

Note that the comment on torch set the macro to -1.

The correct value appears to be 0, see e.g. this commit to the kernel:
https://patchwork.kernel.org/patch/9573661/
---
 gloo/common/linux.cc | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/gloo/common/linux.cc b/gloo/common/linux.cc
index dfd3cb81..f155d440 100644
--- a/gloo/common/linux.cc
+++ b/gloo/common/linux.cc
@@ -31,6 +31,11 @@
 
 #include "gloo/common/logging.h"
 
+#ifndef SPEED_UNKNOWN
+/* some platforms doesn't define speed unknown in headers we included. */
+#define SPEED_UNKNOWN 0
+#endif
+
 namespace gloo {
 
 const std::set<std::string>& kernelModules() {
